src/main/java/hudson/tasks/Ant.java

```diff
@@ -356,7 +357,9 @@ public void setInstallations(AntInstallation... antInstallations) {
 
         @DataBoundConstructor
         public AntInstallation(String name, String home, List<? extends ToolProperty<?>> properties) {
-            super(name, launderHome(home), properties);
+            // JENKINS-57561: while ANT_HOME is an empty string when the tool is configured from UI, if that value is not configured in
+            // yaml files for JCasC, home is null causing a NPE
+            super(name, launderHome(StringUtils.defaultString(home)), properties);
```

Comment:
Maybe it would be worth to include a test for such scenario?

Reply:
Both scenarios are tested in the compatibility test.

This piece of yaml file test the case when `home` is null
```
      - name: "Ant 1.10.5"
        properties:
          - installSource:
              installers:
                - antFromApache:
                    id: "1.10.5"
```
This piece of yaml file test the case when `home` is not null
```
      - home: "/home/ant"
        name: "Ant 1.10.6"
```

Reply:
Wouldn't it be cleaner to update the launderHome method to handle null?
There doesn't seem to be a need to be passing empty strings around to me?

Reply:
launderHome is only used in that constructor and it has been always expecting a non-null string, so another non-null string will be passed to the super constructor. Without being 100% sure about the implication of having a null value there, I'd rather pass an empty string as from the UI. (And I agree that passing a null parameter to super class should be more or less safe)